packages/medusa/src/loaders/defaults.ts

```diff
@@ -126,23 +126,30 @@ export default async ({
 
   await entityManager.transaction(async (manager: EntityManager) => {
     await storeService.withTransaction(manager).create()
+    const profileServiceTx = profileService.withTransaction(manager)
 
     const context = { container, manager, logger }
-    await registerPaymentProvider(context)
-    await registerNotificationProvider(context)
-    await registerFulfillmentProvider(context)
-    await registerTaxProvider(context)
 
-    const profileServiceTx = profileService.withTransaction(manager)
-    await profileServiceTx.createDefault()
-    await profileServiceTx.createGiftCardDefault()
+    await Promise.all([
+      await registerPaymentProvider(context),
+      await registerNotificationProvider(context),
+      await registerFulfillmentProvider(context),
+      await registerTaxProvider(context),
+      await profileServiceTx.createDefault(),
+      await profileServiceTx.createGiftCardDefault(),
```

Comment:
todo: keep the await only for the Promise.all

Reply:
done